bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/SecurityApi.java

```diff
@@ -78,16 +81,26 @@ private Collection<HomeEvent> getEvents(@Nullable Object... params) throws Netat
         throw new NetatmoException("home should not be null");
     }
 
-    public Collection<HomeEvent> getHomeEvents(String homeId) throws NetatmoException {
-        return getEvents(PARAM_HOME_ID, homeId);
+    public List<HomeEvent> getHomeEvents(String homeId, @Nullable ZonedDateTime freshestEventTime) throws NetatmoException {
+        List<HomeEvent> events = getEvents(PARAM_HOME_ID, homeId);
+
+        // we have to rewind to the latest event just after oldestKnown
+        HomeEvent oldestRetrieved = events.get(events.size() - 1);
+        while (freshestEventTime != null && oldestRetrieved.getTime().isAfter(freshestEventTime)) {
+            events.addAll(getEvents(PARAM_HOME_ID, homeId, PARAM_EVENT_ID, oldestRetrieved.getId()));
```

Comment:
Last question: this event (oldestRetrieved) will then be twice in the final list. No bad impact on the binding behaviour ?

Reply:
I will not because the API provides the batch of events just before the one requested.

Reply:
Perfect